<a href="https://colab.research.google.com/github/MichaelBarfuss/Intro-to-Analytics/blob/main/Copy_of_IMSE_441_Midterm_2_(MB).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **IMSE 441 MIDTERM 2**

- Please rename the file as 'IMSE441_Midterm_2_(your initials)
- Use whatever material you have as a reference when solving but please dont discuss with friends or simply copy without logic
- Total exam is 100 marks, of which 40points for in class exam and 60 marks is takehome.


**ALL THE VERY BEST!**

In [ ]:
import pandas as pd
import numpy as np
from statsmodels.stats.power import TTestIndPower

#statistic 
import scipy as scipy
from scipy import stats
import matplotlib.pyplot as plt
import sklearn.linear_model as lm
#from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
from sklearn.metrics import explained_variance_score
%pip install pingouin
import pingouin as pg
import numpy as np
import matplotlib.pyplot as plt
from statsmodels.stats.power import TTestIndPower
!pip install pingouin
import pingouin as pg
from scipy.stats import ttest_ind
from scipy.stats import ttest_rel
from scipy.stats import ttest_1samp
from scipy.stats import ttest_1samp
from scipy.stats import norm


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


### **Question 1:**


# **A**

In [ ]:
oxygen=5.0, 3.4, 3.9, 1.3, 0.2,0.9, 2.7, 3.7, 3.8, 4.1, 1.0, 1.0, 0.8, 0.4, 3.8, 4.5, 5.3, 6.1, 6.9, 6.5

In [ ]:
ttest_1samp(a=oxygen, popmean= 4 )

TtestResult(statistic=-1.5451353441390414, pvalue=0.13880710560932794, df=19)

P-value=0.1388

we do not have enough evidence to support the claim that if the population mean is equal to 4.

# **B**

In [ ]:
#Normality Check
pg.qqplot(oxygen)
plt.show()

# **C**

In [ ]:
power_analysis = TTestIndPower()
power = power_analysis.power(effect_size = 1.18, alpha = 0.01, nobs1= 20, alternative = 'larger')#nobs1= sample size
print('The power for the hypothesis testing is', round(power, 2))

The power for the hypothesis testing is 0.9


# **D**

In [ ]:
power_analysis = TTestIndPower()
sample_size = power_analysis.solve_power(effect_size = 1, alpha = 0.01, power = 0.9, alternative = 'larger')#effect size= d ratio, #alternative= 'two-sided', 'larger', 'smaller'
# Print results
print('The sample size needed for each group is', round(sample_size))

### **Question 2:**





# **A**

In [ ]:
x=63,57,58,62,66,58,61,60,55,62,59,60,58
y=69,65,59,62,61,57,59,60,60,62,61,66,68,66

In [ ]:
ttest_1samp(a=x, popmean= 60 )

TtestResult(statistic=-0.5204164998665357, pvalue=0.6115222287644637, df=13)

In [ ]:
ttest_1samp(a=y, popmean= 60 )

TtestResult(statistic=2.545875386086578, pvalue=0.024382096136580593, df=13)

The P values are very diffenet so I can conclude that these two facilities supply water of different hardnesses.

# **B**

In [ ]:
ttest_ind(a=x, b= y, equal_var= False)

Ttest_indResult(statistic=-2.030125494186018, pvalue=0.05338206244550379)

# **C**

The tests based on different Variances produces a pvalue of 0.5 Which is between the two other tests of 0.024 and 0.611

# **D**

In [ ]:
lmean=np.array(x)
hmean=np.array(y)

In [ ]:
Meanx=lmean.mean()

In [ ]:
Meany=hmean.mean()

In [ ]:
S=Meanx-Meany

In [ ]:
ttest_rel(a=Meany, b= Meanx).confidence_interval(confidence_level=.95)

ConfidenceInterval(low=nan, high=nan)

# **E**

In [ ]:
Before=85.2,79.6, 75.8, 76.2, 91, 89.8, 82, 81.7, 67.3, 68.4, 70, 74, 66.8, 60,94,88.2
After=83.5,78.1,73.2,74,90.2,87,79.9,78.5,64,65.1,67.8,70,64.6,58.6,92.9,88


In [ ]:
ttest_rel(a=Before, b= After).confidence_interval(confidence_level=.99)

ConfidenceInterval(low=1.408408690960369, high=2.916591309039631)

### **Question 3:**


In [ ]:
from google.colab import files
uploaded=files.upload()

In [ ]:
import io

In [ ]:
df =pd.read_excel(io.BytesIO(uploaded['Book1.xlsx']))
df


Month  Temp  Usage/1000
0    Jan.    21      185.79
1    Feb.    24      214.47
2     Mar    32      288.03
3     Apr    47      424.84
4     May    50      454.58
5    June    59      539.03
6    July    68      621.55
7    Aug.    74      675.06
8   Sept.    62      562.03
9    Oct.    50      452.93
10    Nov    41      369.95
11   Dec.    30      273.98

In [ ]:
x = df[['Temp']]
y= df['Usage/1000']
x

Temp
0     21
1     24
2     32
3     47
4     50
5     59
6     68
7     74
8     62
9     50
10    41
11    30

In [ ]:
model = lm.LinearRegression()
model.fit(x,y)

LinearRegression()

In [ ]:
print('Coefficients: ', model.coef_)
print("y-intercept ", model.intercept_)

Coefficients:  [9.20836204]
y-intercept  -6.335501662133765


# **A**

Y=-6.3355+9.2084X



 σ2 = 3.7746


# **B**

Using the line Y=-6.34 + 9.21X plugging in 55 for X



500.1265 Degrees 




# **C**

9.2084 times the 1 degree = 9.2084

# **D**

Using the line Y=-6.3355 + 9.2084X plugging in 47 for X
Y=426.4593
424.84-426.4593
Residual =-1.61751

# **4**

# **5**

Francis Galton

# **6**

In [ ]:
non=10.7,10.7,10.4,10.9,10.5,10.3,9.6,11.1,11.2,10.4
con=9.6,10.4,9.7,10.3,9.2,9.3,9.9,9.5,9.0,10.9

In [40]:
ttest_ind(a=non, b= con, equal_var= True)

Ttest_indResult(statistic=3.356725433186759, pvalue=0.0035122977566031037)

We cannot use paired T test because both groups are differnt.